client/state/sites/reducer.js

```diff
@@ -27,15 +29,18 @@ export function items( state = {}, action ) {
 				[ action.site.ID ]: action.site
 			} );
 		case SERIALIZE:
-			// scrub _events, _maxListeners, and other misc functions
+			//TODO: site objects are being decorated in SitesList in lib/sites/sites-list/index.js
+			//with either lib/site/index.js or lib/site/jetpack.js
```

Comment:
>I also checked what it would take to clean up sites in the redux tree.

It would be nice if we could somehow intercept the received site prior to it being modified in the `sites-list` module. Perhaps [initializing the global instance](https://github.com/Automattic/wp-calypso/blob/master/client/lib/sites-list/index.js) early in the page lifecycle, and having the list module be aware of the Redux store, dispatching the received site upon being received? This seems admittedly awful as a long-term solution, but the long-term solution is to remove the `sites-list` altogether anyways.

Otherwise, maybe we should consider performing the site object cleanup from the `SITE_RECEIVE` reducer handler, so that pages/components making use of sites Redux state don't come to expect these properties and functions to continue to exist.

>Should we move away from this pattern?

Yes. If context-specific behavior is desired, these could probably be reimplemented in the selectors and actions, e.g.

```jsx
if ( isJetpackSite( state, siteId ) ) {
	dispatch( requestAvailableJetpackUpdates( siteId ) );
}
```